ambry-network/src/test/java/com.github.ambry.network/SSLSelectorTest.java

```diff
@@ -203,6 +202,36 @@ public void testEmptyRequest()
     assertEquals("", blockingRequest(connectionId, ""));
   }
 
+  @Test
+  public void testSSLConnect()
+      throws IOException {
+    String connectionId =
+        selector.connect(new InetSocketAddress("localhost", server.port), BUFFER_SIZE, BUFFER_SIZE, PortType.SSL);
+    Assert.assertFalse("Channel should not be ready by now (until handshake completes)",
+        selector.isChannelReady(connectionId));
+    while (!selector.connected().contains(connectionId)) {
+      selector.poll(10000L);
+    }
+    Assert.assertTrue("Channel should have been ready by now ", selector.isChannelReady(connectionId));
+  }
+
+  @Test
+  public void testCloseDuringSslHandshake()
+      throws IOException {
+    String connectionId =
+        selector.connect(new InetSocketAddress("localhost", server.port), BUFFER_SIZE, BUFFER_SIZE, PortType.SSL);
+    Assert.assertFalse("Channel should not be ready by now (until handshake completes)",
```

Comment:
I think the point he is making is that you can't rely on the fact that handshake will not be complete at this point, and the class makes no such guarantees anyway, so this check is not testing the class's functionality.